liota/core/package_manager.py

```diff
@@ -30,4 +30,1104 @@
 #  THE POSSIBILITY OF SUCH DAMAGE.                                            #
 # ----------------------------------------------------------------------------#
 
-# TODO: Implement this module
+import logging
+import inspect
+import imp
+import os
+import fcntl
+import stat
+import re
+import hashlib
+import ConfigParser
+from threading import Thread, Lock
+from Queue import Queue
+from time import sleep
+from abc import ABCMeta, abstractmethod
+
+from liota.lib.utilities.utility import LiotaConfigPath
+
+log = logging.getLogger(__name__)
+
+if __name__ == "__main__":
+    log.warning("Package manager is not supposed to run alone")
+
+    import liota.core.package_manager as actual_package_manager
+    
+    log.debug("MainThread is waiting for interruption signal...")
+    try:
+        while not isinstance(
+                actual_package_manager.package_thread,
+                actual_package_manager.PackageThread
+            ) or actual_package_manager.package_thread.isAlive():
+            sleep(1)
+    except (KeyboardInterrupt, SystemExit):
+        pass
+    finally:
+        if not actual_package_manager.package_message_queue is None:
```

Comment:
(if <condition> is not None) will be much more readable.
http://stackoverflow.com/questions/2710940/python-if-x-is-not-none-or-if-not-x-is-none